source/common/network/io_socket_handle_impl.h

```diff
@@ -0,0 +1,32 @@
+#pragma once
+
+#include "envoy/network/io_handle.h"
+
+namespace Envoy {
+namespace Network {
+
+/**
+ * IoHandle derivative for sockets
+ */
+class IoSocketHandle : public IoHandle {
+public:
+  IoSocketHandle(int fd = -1) : fd_(fd) {}
+
+  // TODO(sbelair2) Call close() in destructor
+  ~IoSocketHandle() {}
+
+  // TODO(sbelair2)  To be removed when the fd is fully abstracted from clients.
+  int fd() const override { return fd_; }
+
+  void close() override { fd_ = -1; }
```

Comment:
I think this is related to @alyssawilk comment above, but can you put an explicit comment here that currently the close syscall happens elsewhere but this will be moved here in the future? This is in addition to potentially also doing this in the destructor.

In general it would make me happier if we fixed the close functionality now, but agree with @alyssawilk that let's get this over the finish line and not add more work on this PR.

Reply:
@mattklein123 Please clarify your point. This is the IoHandle close(), and won't it always be invoked elsewhere, unless explicitly done in the destructor? As for the finish line :-), it's easy enough to just call close from the destructor, since I have another trip around the block with this PR to address your comments. Shall I put it in?

Reply:
My point is eventually this class should own the handle and call `::close()` (syscall close). I would prefer not to go back and forth on that change here and would like to get this merged. Please just add a comment that this is temporary and we will eventually move syscall close into this class. Thanks.